### ISSUE TITLE
predict method

### ISSUE THREAD

**Issue by pwwang** [kevinblighe/PCAtools#16]
Great tools for doing PCA!

But are we able to add a method to allow the reusability of the space being created? Thus to use the pca object to predict the components of the new coming data. For example, with `prcomp`:
```R
p = prcomp(indata)
# reuse the space
p.new = predict(p, newdata = data.new)
```

With `PCAtools`, hopefully, we could do:
```R
p = pca(indata)
p.new = pca(data.new, space = p)
```

Thanks!

**Comment by kevinblighe** [comment]
Hello, thank you for the suggestion. This can be implemented. Can you provide some practical example of how this could be used, even as descriptive text?

**Comment by pwwang** [comment]
One application is, of course, the one I mention in the main post. One would be able to predict with the components being generated in the same space. See an example [here](https://rpubs.com/PandulaP/PCA_for_Predictions).

The other one I can think of is its application in machine learning when we apply PCA for dimension reduction. For example, I have a dataset with covariates, including age and sex, available for 100 patients. Each patient has an expression profile of 30,000 genes. I don't want to put those 30,002 features directly in the model, but just a couple of (i.e. first 2) components of the expression profile together with the covariates. In cross-validation, I can't do PCA to the training and testing sets separately. Instead, I have to apply PCA to the training set to train the model and use the space to get the components for the test set in the same space to validate the model.

**Comment by kevinblighe** [comment]
Thanks - that is a good idea. I will be trying to find time to implement it.

**Comment by kevinblighe** [comment]
Hi @pwwang , I took a look at this and it seems that the `predict()` function interprets the object as prcomp / princomp, and then accesses the relevant variables in this way. I am not sure if you could create a prcomp object manually and then try that (?) Otherwise, I will have to produce an `prcompExport()` function.

- sdev --> sdev 
- center --> FALSE
- scale --> FALSE
- rotation --> loadings
- x --> rotated

**Comment by kevinblighe** [comment]
Hi @pwwang , I have now added a section for this: https://github.com/kevinblighe/PCAtools#make-predictions-on-new-data

Let me know what you think. I added you also to Acknowledgements.

**Comment by pwwang** [comment]
Thanks, this is awesome!
